Changes recommended by Grace to cloud-hosted post (#1574)
OpenLiberty/blogs: posts/2021-07-22-cloud-hosted-guides-post.adoc

---
layout: post
title: "Interactive developer guides running in the cloud"
# Do NOT change the categories section
categories: blog
author_picture: https://avatars3.githubusercontent.com/jamiecoleman92
author_github: https://github.com/GITHUB_USERNAME
seo-title: Interactive developer guides running in the cloud - OpenLiberty.io

> seo-description: We now have a partnership with the IBM Skills Network team to bring you an even better experience by providing you with a full online development environment so that as a developer, you do not have to worry about prerequisites and setting things up. We call these “cloud-hosted guides” and you can access them by going to one of our cloud enabled guides

Reviewer: ```suggestion
seo-description: Open Liberty is providing interactive guides (tutorials hosted in the cloud) for developers and users to learn about cloud-native Java API and technologies like MicroProfile and Jakarta EE without requiring any local prerequisites and setup.
```

> blog_description: "We now have a partnership with the IBM Skills Network team to bring you an even better experience by providing you with a full online development environment so that as a developer, you do not have to worry about prerequisite and setting things up. We call these “cloud-hosted guides” and you can access them by going to one of our cloud enabled guides"

Reviewer: ```suggestion
blog_description: "Open Liberty is providing interactive guides (tutorials hosted in the cloud) for developers and users to learn about cloud-native Java API and technologies like MicroProfile and Jakarta EE without requiring any local prerequisites and setup."
```

open-graph-image: https://openliberty.io/img/twitter_card.jpg
---
= Interactive developer guides running in the cloud
Jamie Coleman <https://github.com/jamiecoleman92>
:imagesdir: /
:url-prefix:
:url-about: /
//Blank line here is necessary before starting the body of the post.

> == Get hands-on with open, cloud-native technologies without the need for any local pre-requisites

Reviewer: ```suggestion
== Get hands-on with open, cloud-native Java technologies without the need for any local pre-requisites
```

> Our team of engineers have been hard at work creating new content for developers to enjoy. We now have 57 different guides On a range of topics and technologies throughout development, build, testing and deployment, designed to help developers get started quickly with Open Liberty and have a great experience with this technology. Excitingly, we now offer the possibility to run our guides in a hosted environment wihtout having to worry about local setup and prerequisits. We’ve named these our ‘cloud-hosted guides’ and you can access them by going to one of our cloud enabled guides on link:{url-prefix}/guides/[Open Liberty] and clicking the “Run in cloud” button in the topright as shown in the image below:

Reviewer: ```suggestion
Our team of engineers have been hard at work creating new content for developers to enjoy. We now have 57 different guides on a range of topics covering development, build, testing and deployment of cloud-native Java applications and microservices. They are designed to help developers get started quickly with learning the technologies and Open Liberty. Excitingly, we now offer the possibility to run our guides in a hosted environment without having to worry about local setup and prerequisites. You can access these cloud-hosted guides by going to our link:{url-prefix}/guides/[guides] page, select a guide, and clicking the “Run in cloud” button in the top right as shown in the following image:
```

image::/img/blog/OL_Cloud-hosted-1.png[Image 1,width=70%,align="center"]

== What guides are available?

> 

Reviewer: We shouldn't need to mention the silent launch.  Let us state simply that 22 guides are now available as interactive guides you can access through a cloud-hosted environment.

Author: Done

Once you are logged in you will then be directed to the landing page for the guide you wish to learn. Simply click on the “Access cloud-hosted guide” button and after a few minutes your environment should be set-up and ready to use.

> == What is the Skills Network and why are we using it with our guides?

Reviewer: We don't need to mention Skills Network.

Author: Done

> The Skills Network is a learning environment that runs through a user’s browser. It runs in containers that are deployed to OpenShift in the IBM Cloud. This allows us not only to teach users how to build great cloud-native applications but also demonstrates that in practice with how the environment is deployed.

Reviewer: Can we re-work this?  We don't need to mention SNL this way.

Author: Done

image::/img/blog/OL_Cloud-hosted-2.png[Image 2,width=70%,align="center"]

This environment gives users access to all the tools required to build cloud-native applications such as Docker, Kubernetes, OpenShift, Maven, Gradle, OpenJ9 JVM and the Open Source Theia IDE.

== Our cloud environment - Skills Network

The environment is quite easy to use with your instructions on the left and your IDE with a terminal on the right. Each guide has different steps with instructions that you can copy straight to your clipboard by clicking on the image:/img/blog/OL_Cloud-hosted-4.png[] button. You can change the font, font size and change the instructions from light to dark mode using the panel above the instructions frame. You can also resize the instruction and IDE frames to suit your screen by dragging on the edge of the frames. 

> The IDE is very similar to VS Code and you can open new terminals by clicking on the terminal tab and selecting `New Terminal`. On the left of the IDE you have the projects button image:/img/blog/OL_Cloud-hosted-5.png[] that will allow you to navigate through your project and open files in the IDE, once you have cloned down the repo for the guide. Maven, Gradle, Docker and Kubernetes are all available for use in the terminal with their normal respective commands. 

Reviewer: We can mention Theia.

Author: Done

Once you have finished a guide you need to log out using the account button in the top right hand corner so that next time you login you will have a clean environment otherwise it will clean-up after 30 minutes of inactivity.

== Help us Help you

You’re welcome to propose new guides by raising an issue and writing an outline for consideration. Especially let us know if you can contribute in some way once the guide has been accepted, whether that’s by writing the guide, writing some code, or both.

> If you see any errors in our existing guides on Skills Network or via the Open Liberty site, feel free to raise an issue against that guide’s GitHub repository (each guide has its own repository, which you can find in the guide text) or alternatively create a PR to fix it. To stay up to date with our new guides, follow us on Twitter (link:https://mobile.twitter.com/OpenLibertyIO[@OpenLibertyio]) or bookmark new guides.

Reviewer: ```suggestion
If you see any errors in our existing guides, feel free to raise an issue against that guide’s GitHub repository (each guide has its own repository, which you can find in the guide text) or alternatively create a PR to fix it. To stay up to date with our new guides, follow us on Twitter (link:https://mobile.twitter.com/OpenLibertyIO[@OpenLibertyio]) or bookmark new guides.
```
